conda/recipes/cudf/meta.yaml

```diff
@@ -64,8 +64,7 @@ requirements:
     - rapids-build-backend >=0.3.0,<0.4.0.dev0
     - scikit-build-core >=0.10.0
     - dlpack >=0.8,<1.0
-    # TODO: Change to `2.0` for NumPy 2
-    - numpy 1.23
+    - numpy 2.0
```

Comment:
Thanks, good to get this done.  But I suspect we can just remove NumPy as a host requirement for cudf?  (at least as a direct requirement)

Reply:
Unfortunately as Arrow is currently needed during the build (and Arrow uses NumPy headers in some cases), there is a NumPy build dependency. Please see this comment: https://github.com/rapidsai/cudf/pull/15165#issuecomment-1968090495

Once Arrow is dropped as a build dependency, we can remove NumPy from build dependencies too: https://github.com/rapidsai/cudf/issues/15193#issuecomment-2240801140

Reply:
Ah right, sorry I forgot.  For that it should be OK to build with 1.x, but I agree it is better to just change it!

Reply:
Ah sorry I should have mentioned the second issue, which is we need to [relax the NumPy pinning in RMM]( https://github.com/rapidsai/rmm/blob/975c911ad59219e98b8aed1e9959120a2c877d06/dependencies.yaml#L281 )

For example we see this error [on CI]( https://github.com/rapidsai/cudf/actions/runs/10360138990/job/28678121967?pr=16300#step:9:153 ):

```python
  ERROR: Cannot install numpy==2.0.*, rmm-cu11==24.10.0a21 and rmm-cu11==24.10.0a22 because these package versions have conflicting dependencies.

  The conflict is caused by:
      The user requested numpy==2.0.*
      rmm-cu11 24.10.0a22 depends on numpy<2.0a0 and >=1.23
      The user requested numpy==2.0.*
      rmm-cu11 24.10.0a21 depends on numpy<2.0a0 and >=1.23
```

If we do that, we can build with NumPy 2 here (even if we are still using NumPy 1 at runtime)

Reply:
Thanks to Sebastian we are now building RMM to allow NumPy 2: https://github.com/rapidsai/rmm/pull/1650

Will update this PR once there are RMM packages available

Reply:
> Once Arrow is dropped as a build dependency, we can remove NumPy from build dependencies too: https://github.com/rapidsai/cudf/issues/15193#issuecomment-2240801140

Just noting that #16640 is up for review and all its CI is passing, I suspect it'll be merged soon.

This PR and that one will have some conflicts... which is good, because I think one should be tested again after the other is merged.